sky/resources.py

```diff
@@ -601,18 +570,27 @@ def less_demanding_than(self,
         # self <= other
         return True
 
-    def is_launchable_fuzzy_equal(self, other: 'Resources') -> bool:
-        """Whether the resources are the fuzzily same launchable resources."""
-        assert self.cloud is not None and other.cloud is not None
-        if not self.cloud.is_same_cloud(other.cloud):
-            return False
-        if self._instance_type is not None or other.instance_type is not None:
-            return self._instance_type == other.instance_type
-        if self._region is not None or other.region is not None:
-            return self._region == other.region
-        if self._zone is not None or other.zone is not None:
-            return self._zone == other.zone
-        return self.accelerators.keys() == other.accelerators.keys()
+    def should_be_blocked_by(self, blocked: 'Resources') -> bool:
+        """Whether this Resources matches the blocked Resources.
+
+        If a field in `blocked` is None, it should be considered as a wildcard
+        for that field.
+        """
+        is_matched = True
+        if (blocked.cloud is not None and
+                not self.cloud.is_same_cloud(blocked.cloud)):
+            is_matched = False
+        if (blocked.instance_type is not None and
+                self.instance_type != blocked.instance_type):
+            is_matched = False
+        if blocked.region is not None and self._region != blocked.region:
+            is_matched = False
+        if blocked.zone is not None and self._zone != blocked.zone:
+            is_matched = False
+        if (blocked.accelerators is not None and
+                self.accelerators != blocked.accelerators):
```

Comment:
nit: why did we check `accelerators.keys()` but now `accelerators`?

Reply:
While I believe it will not matter in practice, I feel the current one is a little bit more accurate because otherwise `V100:1` can be blocked by `V100:4`.